sdk-server-undertow/src/main/java/com/manywho/sdk/services/servers/undertow/UndertowServer.java

```diff
@@ -63,10 +79,80 @@ public void start() throws Exception {
         start("/");
     }
 
+    /**
+     * Start a V2 instance of the service using the built-in Jetty container on a specified https port
+     * 
+     * @param path the path to run the service from
+     * @param httpsPort the port to run the https service on
+     * @param keyStore the keystore containing the server certificate
+     * @param trustStore the truststore containing trusted certificates. If null, client certificate authentication will not be requested from clients
+     * @param keyStorePassword the password for the keystore
+     * @param trustStorePassword the password for the truststore
+     */
+    public void start(String path, int httpsPort, InputStream keyStore, InputStream trustStore, String keyStorePassword, String trustStorePassword) {
+
+        ServiceApplication serviceApplication = new ServiceApplication();
+
+        for (Module module : modules) {
+            serviceApplication.addModule(module);
+        }
+
+        serviceApplication.initialize(application.getPackage().getName(), true, true);
+
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(buildKeyManagerFromStream(keyStore, keyStorePassword), buildTrustManagerFromStream(trustStore, trustStorePassword), null);
+
+            boolean clientAuthEnabled = trustStore != null;
+
+            Undertow.Builder serverBuilder = Undertow.builder()
+                .addHttpsListener(httpsPort, "0.0.0.0", sslContext)
+                .setSocketOption(Options.SSL_CLIENT_AUTH_MODE, clientAuthEnabled ? SslClientAuthMode.REQUIRED : SslClientAuthMode.NOT_REQUESTED);
+
+            server = new UndertowJaxrsServer();
+            server.start(serverBuilder);
+            server.deploy(serviceApplication, path);
+
+            LOGGER.info("Service started on https://0.0.0.0:{} - Client auth enabled: {}", httpsPort, clientAuthEnabled);
+            LOGGER.info("Stop the service using CTRL+C");
+        } catch (Exception ex) {
+            LOGGER.error("Unable to start the server", ex);
+        }        
+    }
+
     @Override
     public void stop() {
         if (server != null) {
             server.stop();
         }
     }
+
+    private KeyManager[] buildKeyManagerFromStream(InputStream storeStream, String storePassword) throws NoSuchAlgorithmException, CertificateException, IOException, KeyStoreException, UnrecoverableKeyException {
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        keystore.load(storeStream, storePassword.toCharArray());
+        System.out.println("Keystore loaded with keys: " + keystore.size());
```

Comment:
Just a minor thing - but can we do it with a logger? As then it's easier to turn off such output

Reply:
Done